### DISCUSSION TITLE
Welcome to AeroSandbox Discussions! **PLEASE READ THIS BEFORE CREATING A NEW DISCUSSION.**

### DISCUSSION THREAD

**Discussion by peterdsharpe** [peterdsharpe/AeroSandbox#61] [Announcements]
Welcome to our Discussions forum, where you can ask questions, get help with any AeroSandbox-related issues, or help other users.

## Purpose

This Discussions forum is generally intended for usage-related questions - if you find something that you think is a bug in the framework itself, or you'd like to request a new feature, that might be [better-suited as an issue ticket](https://github.com/peterdsharpe/AeroSandbox/issues).

## **Discussion Guidance (Read this!)**

The easier you can make it for someone to help you, the more likely you are to receive timely help. If you have questions about behavior in a particular piece of code, **include a minimal, self-contained, properly-formatted example of the code that you're having issues with**. This means:

* **Self-contained**: the script should run with no extra imports or reference to variables outside of the script
* **Minimal**: strip away any extraneous content that doesn't affect the topic of your questions. If I can remove any lines of code from the example and still reproduce the error, it's not minimal.
* **Properly-formatted**:
    * Format your code using something like [Ruff](https://github.com/astral-sh/ruff), and use descriptive and helpful variable names and comments.
    * On GitHub: [**wrap any code in Markdown code fences**](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code) (i.e., surround your code with ` ``` ` on both sides; ideally starting with ` ```python ` so you get syntax highlighting).
    * Never include screenshots of code - include the actual code as text.

If you are not using the latest public AeroSandbox version (compare [here](https://pypi.org/project/AeroSandbox/) and `asb.__version__`; these should match), please mention that.

It is expected that this will take some effort on your part, but in return, you're much more likely to get a timely answer.

**Comment by Foilit** [comment]
Im interested in hydrofoils and optimization, can Aerosandbox and Neuralfoil be used for foils in water? Thanks.

**Reply by ArthurLaneri05** [reply]
Hi, I think that with few edits it could do the job. 
However, keep in mind that NeuralFoil doesn't provide any data about cavitation (maybe you can get around that by limiting the Cp_min of the airfoil), and that the density of the fluid referenced by the analyses isn't directly input as an argument (but I might be wrong on this), as it is derived from an atmospheric model.
But as I said before, nothing too crazy to get around.

**Reply by peterdsharpe** [reply]
Hi @Foilit ! Yes, in general it can be applied to hydrofoils, depending on your specific interests. @ArthurLaneri05 's suggestions are correct (to get around the density issue, you should define your own class that extends `asb.Atmosphere` with your desired fluid properties (e.g., `class WaterAtmosphere(asb.Atmosphere):`). This can then be fed into any `asb.OperatingPoint` instance, and the new fluid properties would apply.

You may be interested in checking out a recent Master's thesis that used AeroSandbox for hydrofoil optimization here: https://www.politesi.polimi.it/handle/10589/209472 I think they also developed free-surface corrections and cavitation models for use with AeroSandbox, if memory serves correctly.

**Reply by Foilit** [reply]
Thanks!
